pkgs/by-name/pk/pkg-config-unwrapped/package.nix

```diff
@@ -26,7 +26,8 @@ stdenv.mkDerivation rec {
   # http://bugs.freedesktop.org/show_bug.cgi?id=4738, migrated to
   # https://gitlab.freedesktop.org/pkg-config/pkg-config/issues/28
   patches =
-    lib.optional (!vanilla) ./requires-private.patch
+    [ ./gcc-15.patch ]
```

Comment:
Since GCC is not the current version yet, let's do `stdenv.cc.isGNU && lib.versionAtLeast stdenv.cc.cc.version "15.0"`

Reply:
I don't think variable rename would hurt the gcc-14 build, but I am fine either way.

As I mentioned, I don't have CPU to do proper full-world rebuild, but
```
:b pkg-config-unwrapped.override { stdenv = gcc15Stdenv; }
```

works here and fails on master.

Reply:
Re-targeted to master as you requested, thought.

Reply:
Sometimes, staging's pace can change.

Reply:
@RossComputerGuy because this was brought up in #421115 again: I think saving rebuilds here is the wrong call.

There's zero rush to do it, but conditionals make this more complicated and I don't see why this would fail with gcc14.